src/XrdTpc/XrdTpcTPC.hh

```diff
@@ -76,6 +79,8 @@ private:
         int tpc_status;
         unsigned int streams;
         bool isIPv6;
+        // The packet manager lifetime is the one of the TPCLogRecord
+        std::unique_ptr<PMarkManager> pmarkManager;
```

Comment:
I think using
```suggestion
        MarkManager pmarkManager;
```
here would allow you to remove a lot of the `if(rec.pmarkManager)` statements you have. You'd just then internally have to deal with null pointers in the `PMarkManager` class itself.